git/repo.go

```diff
@@ -39,6 +39,7 @@ func (r Repo) Clone() (path string, err error) {
 
 	// Hack, while it's not possible to mount a secret with a
 	// particular mode in Kubernetes
+	// FIXME this fails if the key is mounted as read-only.
```

Comment:
Absolutely true --  if it's mounted as read-only, we can't proceed. We could at least test whether the permissions are correct first, and provide a reasonable error if they aren't and we can't do anything about it.